pkg/apis/agones/v1/gameserver.go

```diff
@@ -532,24 +554,26 @@ func (gs *GameServer) Pod(sidecars ...corev1.Container) (*corev1.Pod, error) {
 	}
 
 	gs.podObjectMeta(pod)
-
-	i, gsContainer, err := gs.FindGameServerContainer()
-	// this shouldn't happen, but if it does.
-	if err != nil {
-		return pod, err
-	}
+	pod.Spec.Containers = append(pod.Spec.Containers, sidecars...)
 
 	for _, p := range gs.Spec.Ports {
 		cp := corev1.ContainerPort{
 			ContainerPort: p.ContainerPort,
 			HostPort:      p.HostPort,
 			Protocol:      p.Protocol,
 		}
-		gsContainer.Ports = append(gsContainer.Ports, cp)
+		portApplied := false
+		for j, container := range pod.Spec.Containers {
+			if container.Name == p.ContainerName {
```

Comment:
Probably also makes sense to update and rename `ApplyToPodGameServerContainer` to use with this logic, since it's almost the same, except you would want to pass in a name, rather than use the gs.Spec.Container by default.